docs/architecture/testing/invasive-testing.md

```diff
@@ -0,0 +1,358 @@
+# Mbed TLS invasive testing strategy
+
+## Introduction
+
+In Mbed TLS, we use black-box testing as much as possible: test the documented behavior of the product, in a realistic environment. However this is not always sufficient.
+
+The goal of this document is to identify areas where black-box testing is insufficient and to propose solutions.
+
+This is a test strategy document, not a test plan. A description of exactly what is tested is out of scope.
+
```

Comment:
Suggestion: while I find the document to be generally very readable and well-structured, perhaps this could still be improved a bit by outlining the top-level structure here with a quick explanation, so that we know what to expect.

Suggested text: "This document starts with a section summarizing what to do and what not to do when testing, and the other sections expand on the rationale by first clarifying the goals, then exploring possible approaches, and finally listing solutions to specific problems."

Reply:
I added a short overview. I'm open to suggestions of better section names.